packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts

```diff
@@ -184,45 +178,26 @@ export default class ImageEdit implements EditorPlugin {
      */
     onPluginEvent(e: PluginEvent) {
         switch (e.eventType) {
-            case PluginEventType.MouseDown:
-                this.setEditingImage(null);
-                break;
-
-            case PluginEventType.MouseUp:
-                const target = e.rawEvent.target;
+            case PluginEventType.SelectionChanged:
                 if (
-                    e.isClicking &&
-                    e.rawEvent.button == 0 &&
-                    safeInstanceOf(target, 'HTMLImageElement') &&
-                    target.isContentEditable &&
-                    matchesSelector(target, this.options.imageSelector)
+                    e.selectionRangeEx &&
+                    e.selectionRangeEx.type === SelectionRangeTypes.ImageSelection
                 ) {
-                    this.setEditingImage(target, ImageEditOperation.ResizeAndRotate);
-                }
-
-                break;
-
-            case PluginEventType.KeyDown:
-                const key = e.rawEvent.which;
-                if (key == Keys.DELETE || key == Keys.BACKSPACE) {
-                    // Set current editing image to null and select the image if any, and do not prevent default of the event
-                    // so that browser will delete the selected image for us
-                    this.setEditingImage(null, true /*selectImage*/);
-                } else if (key == Keys.ESCAPE && this.image) {
-                    // Press ESC should cancel current editing operation, resume back to original edit info
-                    this.editInfo = getEditInfoFromImage(this.image);
+                    this.setEditingImage(
+                        e.selectionRangeEx?.image,
```

Comment:
You already did null check of selectionRangeEx in line 183, no need to add "?" here